src/Microsoft.Health.Fhir.Api/Features/ActionResults/ResourceActionResult.cs

```diff
@@ -70,16 +71,27 @@ public override async Task ExecuteResultAsync(ActionContext context)
 
             HttpResponse response = context.HttpContext.Response;
 
-            if (fhirContext.GetMissingResourceCode() != null)
+            if (fhirContextAccessor.GetMissingResourceCode() != null)
             {
-                response.StatusCode = (int)fhirContext.GetMissingResourceCode().Value;
+                response.StatusCode = (int)fhirContextAccessor.GetMissingResourceCode().Value;
             }
             else if (StatusCode.HasValue)
             {
                 response.StatusCode = (int)StatusCode.Value;
             }
 
+            Dictionary<string, StringValues> finalHeaders = new Dictionary<string, StringValues>();
             foreach (KeyValuePair<string, StringValues> header in Headers)
+            {
+                finalHeaders.TryAdd(header.Key, header.Value);
```

Comment:
Why do we need to add these to the intermediate `finalHeaders` dictionary first instead of each source adding to `response.Headers` themselves?

Reply:
Changed it.